Property '_WORKLET' doesn't exist

### Description

Property '_WORKLET' doesn't exist

### Steps to reproduce

1. Initiate RN project with version "0.78.0"
2. install "react-native-reanimated" with version "4.0.0-beta.2"
3. On any usage of this package the issue triggers

### Snack or a link to a repository

https://snack.expo.io/

### Reanimated version

4.0.0-beta.2

### React Native version

0.78.0

### Platforms

iOS

### JavaScript runtime

Hermes

### Workflow

React Native

### Architecture

Fabric (New Architecture)

### Build type

Debug app & dev bundle

### Device

iOS simulator

### Device model

iPhone 16 Pro

### Acknowledgements

Yes

> Yeahh i invited you to my repo

#### Problems with `npm` installation

I couldn't get dependencies installed right with `npm`. I had to remove your `package-lock.json` and install node modules with `--force` flag to, at least, finish installation without error. In the end, the app showed errors after building (in my case related to the `react-native-screens` library), because dependencies weren't installed properly.

<details>
<summary>Stack trace of error I was getting</summary>

```sh
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @rneui/themed@4.0.0-rc.8
npm ERR! Found: @rneui/base@4.0.0-rc.8
npm ERR! node_modules/@rneui/base
npm ERR!   @rneui/base@"^4.0.0-rc.8" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @rneui/base@"4.0.0-rc.7" from @rneui/themed@4.0.0-rc.8
npm ERR! node_modules/@rneui/themed
npm ERR!   @rneui/themed@"^4.0.0-rc.8" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: @rneui/base@4.0.0-rc.7
npm ERR! node_modules/@rneui/base
npm ERR!   peer @rneui/base@"4.0.0-rc.7" from @rneui/themed@4.0.0-rc.8
npm ERR!   node_modules/@rneui/themed
npm ERR!     @rneui/themed@"^4.0.0-rc.8" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /Users/mateu/.npm/_logs/2025-03-05T21_14_10_084Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: /Users/mateu/.npm/_logs/2025-03-05T21_14_10_084Z-debug-0.log
```
</details>

#### Installation with `yarn`

I tried using `yarn`. After finishing these steps, the app started running without errors:
1. Remove `package-lock.json`
2. Install dependencies with `yarn` package manager (you can just use the `yarn` command)
3. Run `yarn start` to start the metro bundler
4. Build the app (either from terminal or from Android Studio / Xcode - for iOS)

#### Example recording

https://github.com/user-attachments/assets/b6356acf-58e4-4a45-8bb8-28149b83273c

Please let me know if it helps. I don't know why dependencies cannot be installed properly with `npm` but I am using `yarn` daily as it seems to deal better with dependency resolution.

--- a/packages/react-native-reanimated/src/common/constants/platform.ts
+++ b/packages/react-native-reanimated/src/common/constants/platform.ts
@@ -13,7 +13,8 @@ export const IS_ANDROID: boolean = /* @__PURE__ */ (() =>
   Platform?.OS === 'android')();
 export const IS_IOS: boolean = /* @__PURE__ */ (() => Platform?.OS === 'ios')();
 export const IS_WEB: boolean = Platform?.OS === 'web';
-export const IS_JEST: boolean = !!process.env.JEST_WORKER_ID;
+export const IS_JEST: boolean =
+  typeof globalThis.jest !== 'undefined' || process.env.NODE_ENV === 'test';
 /** @knipIgnore */
 export const IS_WINDOWS: boolean = Platform?.OS === 'windows';
 

--- a/packages/react-native-worklets/src/platformChecker.ts
+++ b/packages/react-native-worklets/src/platformChecker.ts
@@ -1,3 +1,4 @@
 'use strict';
 
-export const IS_JEST: boolean = !!process.env.JEST_WORKER_ID;
+export const IS_JEST: boolean =
+  typeof globalThis.jest !== 'undefined' || process.env.NODE_ENV === 'test';

